packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -141,10 +141,19 @@ export function useVirtualizer<T extends object, V, W>(props: VirtualizerOptions
       } else {
         virtualizer.scrollToItem(focusedKey, {duration: 0});
       }
+
+      if (modality === 'keyboard' && ref.current.contains(document.activeElement)) {
+        // Wait till virtualizer scrolls the focused key into view before checking if we need to scroll it into the viewport
+        // Specifically need to do this for the column headers since they aren't part of the scrollable table body and their position
+        // syncing is a bit delayed compared to when scrollToItem is fired.
+        requestAnimationFrame(() => {
```

Comment:
don't forget to cleanup the raf in case we unmount mid raf
I think there were a couple others in the other files, make sure they are all cleaned up